### PR TITLE
Modularize stores

### PR DESCRIPTION
**Pull request by apechimp**
This breaks the redis dependency out of dipswitch in preparation for the addition of a postgres store. Note that this change will require revving the major semver.

Additionally, the tests have been made require-able so that store implementations may use them to check their implementations.

### CODE REVIEW

**Comment by apechimp** [comment]
CR @lanetix/engineers: this is relevant to lanetix/issues#1892

File: test/index.js [hunk 1]
```diff
@@ -115,525 +113,535 @@ function userHasNoFeatures(t, d, user) {
   };
 }
 
-test('default configuration', function (t) {
+function runTests(dipswitchOptions) {
```
Comment (apechimp) [inline test/index.js]: From here until the end of the function the only changes are indentation changes.

File: test/index.js [hunk 2]
```diff
@@ -115,525 +113,535 @@ function userHasNoFeatures(t, d, user) {
...
+          .then(userHasNoFeatures(t, d, user3))
+          .then(function () {
+            return d.tenant({id: user1.tenant}).group('group2').disable('TEST_FEATURE_1');
+          })
+          .then(userHasNoFeatures(t, d, user1));
+      });
     });
   });
-});
+}
```
Comment (apechimp) [inline test/index.js]: The __very__ noisy whitespace diff ends here.

**Comment by tone81** [comment]
:bread:

**Pull request merged**